src/Shared/Components/ConfirmationModal/ConfirmationModal.tsx

```diff
@@ -0,0 +1,167 @@
+import { ButtonHTMLAttributes, ChangeEvent, useCallback, useEffect, useState } from 'react'
+import { AnimatePresence, motion } from 'framer-motion'
+import { CustomInput, useRegisterShortcut, UseRegisterShortcutProvider } from '@Common/index'
+import { ComponentSizeType } from '@Shared/constants'
+import { ConfirmationModalProps } from './types'
+import { getPrimaryButtonStyleFromVariant, getConfirmationLabel, getIconFromVariant } from './utils'
+import { Button, ButtonStyleType, ButtonVariantType } from '../Button'
+import './confirmationModal.scss'
+import { Backdrop } from '../Backdrop'
+
+const ConfirmationModal = ({
+    title,
+    subtitle,
+    Icon,
+    variant,
+    buttonConfig,
+    customInputConfig,
+    children,
+    showConfirmationModal,
+    handleClose,
+}: ConfirmationModalProps) => {
+    const { registerShortcut, unregisterShortcut } = useRegisterShortcut()
```

Comment:
Please try to add some gap and maintain order between hook, states and variables